src/cpp/rtps/transport/TCPChannelResourceSecure.h

```diff
@@ -87,6 +88,10 @@ class TCPChannelResourceSecure : public TCPChannelResource
         return secure_socket_;
     }
 
+    void update_channel(
```

Comment:
This method could receive another channel, and take all the required information from there